www/src/components/layout/WithApplicationUpdate.js

```diff
@@ -1,48 +1,55 @@
-import { useCallback, useContext, useState } from 'react'
+import {
+  useCallback, useContext, useEffect, useState,
+} from 'react'
 
-import * as serviceWorker from '../../serviceWorker'
+import * as serviceWorker from '../../serviceWorkerRegistration'
 
 import { PluralConfigurationContext } from '../login/CurrentUser'
 
 const COMMIT_KEY = 'git-commit'
 
 export const DEFAULT_COMMIT = 'plural-default-commit'
 
-export const getCommit = () => localStorage.getItem(COMMIT_KEY) || DEFAULT_COMMIT
-export const setCommit = sha => localStorage.setItem(COMMIT_KEY, sha)
+export const getCommit = () => sessionStorage.getItem(COMMIT_KEY) || DEFAULT_COMMIT
+export const setCommit = sha => sessionStorage.setItem(COMMIT_KEY, sha)
 
 function WithApplicationUpdate({ children }) {
-  const [open, setOpen] = useState(true)
+  const [time, setTime] = useState(Date.now())
   const config = useContext(PluralConfigurationContext)
   const commit = getCommit()
-
+  const [stale, setStale] = useState(commit !== config.gitCommit)
   const reloadApplication = useCallback(() => {
-    console.log('reloading')
-    if (process.env.NODE_ENV === 'production') {
-      const promise = serviceWorker.unregister() || Promise.resolve('done')
-
-      promise.then(() => {
-        setCommit(config.gitCommit)
-        window.location.reload()
-      })
-    }
-    else {
+    const promise = serviceWorker.unregister() || Promise.resolve('done')
+
+    setStale(false)
+
+    promise.then(() => {
       setCommit(config.gitCommit)
-      setOpen(false)
-    }
-  }, [config, setOpen])
+      window.location.reload()
+    })
+  }, [config])
+  const reloadOnStale = () => (stale ? reloadApplication() : null)
+
+  // force rerender every 60s to check if version hasn't changed
+  useEffect(() => {
```

Comment:
I am not 100% sure if this is needed. The config that is part of the plural context should trigger rerender here. I did use it to trigger the update logic when `sessionStorage` entry was changed by me manually. It was easier than mocking the graphql API response to trigger it. In the end, it shouldn't have any real impact on performance as it is quite small and triggered every 60s.